helios-client/src/main/java/com/spotify/helios/common/descriptors/Secrets.java

```diff
@@ -0,0 +1,179 @@
+ * Configures automatic secrets procurement for a container.
+ *
+ * <pre>
+ *   {
+ *     "source": "Talos",
+ *     "path": "/secrets",
+ *     "keypairFromHost": {
+ *       "certificate": "/path/to/pem/cert/on/host",
+ *       "privateKey": "/path/to/pem/key/on/host"
+ *     }
+ *   }
+ * </pre>
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Secrets extends Descriptor {
+  private final Source source;
+  private final String path;
+  private final KeyPair keyPairFromHost;
+
+  /**
+   * Automatic secret procurement configuration.
+   *
+   * @param source The source from which to procure secrets for this job.
+   * @param path The path inside the container at which to mount the secrets for this job.
+   * @param keyPairFromHost Optional keypair to use to authenticate to the secret source on behalf
+   *                        of this container.
+   */
+  public Secrets(@JsonProperty("source") final Source source,
+                 @JsonProperty("path") final String path,
+                 @JsonProperty("keyPairFromHost") @Nullable final KeyPair keyPairFromHost) {
+    // TODO(negz): Support tags. Inherit from agent labels, but allow override?
+    this.source = source;
+    this.path = path;
+    this.keyPairFromHost = keyPairFromHost;
+  }
+
+  /**
+   * Create a new secret procurement configuration.
+   *
+   * @param source The secret source (i.e. Talos)
+   * @param path The path inside the container at which to mount the secrets for this job.
+   * @param certPath The path on the host at which to find a certificate for secret authentication.
+   * @param keyPath The path on the host at which to find a private key for secret authentication.
+   * @return A new secret procurement configuration.
+   */
+  public static Secrets create(final Source source, final String path,
+                               final String certPath, final String keyPath) {
+    return new Secrets(source, path, new KeyPair(certPath, keyPath));
+  }
+
+  public Source getSource() {
+    return this.source;
+  }
+
+  public String getPath() {
+    return this.path;
+  }
+
+  @Nullable
+  public KeyPair getKeyPairFromHost() {
+    return this.keyPairFromHost;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    final Secrets that = (Secrets) o;
+
+    return Objects.equals(this.source, that.source)
+           && Objects.equals(this.path, that.path)
+           && Objects.equals(this.keyPairFromHost, that.keyPairFromHost);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(source, path, keyPairFromHost);
+  }
+
+  @Override
+  public String toString() {
+    return "Secrets{"
+           + "source=" + source
+           + ", path=" + path
+           + ", keyPairFromHost=" + keyPairFromHost
+           + "} " + super.toString();
+  }
+
+  public enum Source {
+    UNKNOWN,
+    TALOS
+  }
```

Comment:
if all possible names for the sources are defined within this class, then this means that the only secret providers that could be used have to be defined in the helios source code. in other words it is not very open for extension or pluggable. I think the plugin approach used by the service registration code would work great here to avoid tight coupling.